torchao/quantization/qat/api.py

```diff
@@ -148,18 +150,26 @@ def __post_init__(self):
         all_step_values = [s.value for s in QATStep]
         if self.step not in all_step_values:
             raise ValueError(f"`step` must be one of {all_step_values}")
-        if self.base_config is None and self.weight_config is None:
-            raise ValueError(
-                "One of `base_config` or `weight_config` must be specified"
-            )
         if self.base_config is not None and self.activation_config is not None:
             raise ValueError(
                 "Cannot specify both `base_config` and `activation_config`"
             )
         if self.base_config is not None and self.weight_config is not None:
             raise ValueError("Cannot specify both `base_config` and `weight_config`")
-        if self.base_config is None and self.step == "convert":
-            raise ValueError("`base_config` must be specified in the convert step")
+        if (
+            self.step == QATStep.PREPARE
+            and self.base_config is None
+            and self.weight_config is None
```

Comment:
@andrewor14 Would it be possible to allow base_config=None and weight_config=None? This would support PARQ's activation-only quantization use case in #2743.

Reply:
yeah let's do that separately in your PR